src/parallel/parallel.cpp

```diff
@@ -163,7 +186,10 @@ ParallelRobinHoodHashTable::find_next_index_lock(ThreadManager &manager,
 ThreadManager &
 ParallelRobinHoodHashTable::get_thread_lock_manager()
 {
-    // TODO
+    std::thread::id t_id = std::this_thread::get_id();
+    assert(this->thread_managers_.contains(t_id) &&
+           "thread manager for current thread DNE");
+    return this->thread_managers_.at(t_id);
```

Comment:
I wonder if we can just have:
```c++
ThreadManager &
ParallelRobinHoodHashTable::get_thread_lock_manager()
{
    static thread_local ThreadManager manager(this);
    return manager;
}
```

Then we don't have to worry about add and remove.

Non-blocking.

Reply:
The reason for not is because we want it both *per thread* and *per hash table*. So we need to associate it with both.

I will add this as a comment.

Reply:
Oh wait sorry, I'm looking at this more closely. So each thread has a ThreadManager, which maintains a list of all the ParallelRobinHoodHashTables it manages.

I like the cleaner setup/destroy semantics. Maybe something to look into for the future.

Reply:
Created new MR #19 to address this in case it is a breaking change.